webpack/src/Extends/Host/PuppetTab/SubTabs/YamlTab/index.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import PropTypes from 'prop-types';
+
+import Skeleton from 'react-loading-skeleton';
+import EmptyState from 'foremanReact/components/common/EmptyState/EmptyStatePattern';
+import { STATUS } from 'foremanReact/constants';
+import { EmptyStateIcon } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { global_danger_color_200 as dangerColor } from '@patternfly/react-tokens';
+import { translate as __ } from 'foremanReact/common/I18n';
+import './YamlTab.scss';
+
+const YamlTab = ({ hostName }) => {
+  const params = { params: { name: hostName, format: 'yml' } };
+  const url = `${window.location.origin.toString()}/foreman_puppet/hosts/${hostName}/externalNodes`;
+  const { response: yaml, status } = useAPI('get', url, params);
+
+  if (status === STATUS.PENDING) {
+    return <Skeleton count={5} />;
+  }
+
+  if (status === STATUS.ERROR) {
+    const icon = (
+      <EmptyStateIcon icon={ExclamationCircleIcon} color={dangerColor.value} />
+    );
+    return (
+      <EmptyState
+        header={__('Error!')}
+        icon={icon}
+        description={yaml?.response?.data?.error?.message}
```

Comment:
I guess it makes more sense to keep it as `response` instead of `yaml` as it's error

Reply:
oh yeah, sorry... I meant if we could rename it from `yaml` `response`